### DISCUSSION TITLE
Your Solution Ideas

### DISCUSSION THREAD

**Discussion by getlost01** [getlost01/gfg-potd#10] [Solution Ideas]
Please provide your solution Ideas here...

Very Welcome `<Happy Coding/>`

**Comment by SHRbharat** [comment]
For potd , 10 Nov 2023 

`string printMinNumberForPattern(string S){
    int n = S.length();
    string result;

    stack<int> st;
    int current = 1;

    for (int i = 0; i <= n; ++i) {
        st.push(current++);

        if (i == n || S[i] == 'I') {
            while (!st.empty()) {
                result += to_string(st.top());
                st.pop();
            }
        }
    }

    return result;
}`

**Reply by getlost01** [reply]
Yep, that is also a good solution. Thanks for sharing your idea of using the stack for this question.

**Comment by SHRbharat** [comment]
For potd 14 Nov,

`
    bool areRotations(string s1,string s2){
        string concat = s1 + s1;
        return concat.find(s2) != string::npos;
    }
`

**Comment by SHRbharat** [comment]
for POTD - 26 NOV ; it's mentioned to not use loops.
> here's my solution for this (please consider )

```c++
class Solution{
public:
    bool decreasing = true; 
    void patternHelper(int current, int initial,vector<int>& result) {
        /*
        cout<<"function started  : ";
        for(int u: result){
            cout<<u<<" ";
        }
        cout<<endl;
        */
        if(current > initial){
            return;
        }else if(decreasing){
            result.push_back(current);
            if(current <= 0){
                decreasing = false;
                patternHelper(current+5,initial,result);
            }else{
                patternHelper(current-5,initial,result);
            }
        }else{
            result.push_back(current);
            patternHelper(current+5,initial,result);
        }
    }
    
    vector<int> pattern(int N){
        // code here
        vector<int> result;
        patternHelper(N,N,result);
        
        return result;
    }
};
```

**Reply by getlost01** [reply]
Oh, I see it now. To be honest, I initially noticed its easy tag and question title, so I wrote the code without even seeing the whole problem statement. 

I really appreciate that you've shared your solution. Thanks!

**Reply by getlost01** [reply]
Now, I've updated the code by using a recursive approach.

Your insights and contribution are really appreciated.

**Comment by mayad3052** [comment]
08-02-2024
POTD

class Solution{
  public:
    /*You are required to complete this method*/
    bool checkUtil(Node *root, int level, int* leafLevel)
    {
        // Base case
        if (root == NULL) return true;
  
        // If a leaf node is encountered
        if (root->left == NULL && root->right == NULL)
        {
            // When a leaf node is found first time
            if (*leafLevel == 0)
            {
                *leafLevel = level; // Set first found leaf's level
                return true;
            }
  
            // If this is not first leaf node, compare its level with
            // first leaf's level
            return (level == *leafLevel);
        }
  
        // If this node is not leaf, recursively check left and right subtrees
        return checkUtil(root->left, level+1, leafLevel) && checkUtil(root->right, level+1, leafLevel);
    }
  
    bool check(Node *root)
    {
        int level = 0, leafLevel = 0;
        return checkUtil(root, level, &leafLevel);
    }
};

**Reply by getlost01** [reply]
I liked your logic 👍, thanks
